modules/glfw/src/glfwmodule.cpp

```diff
@@ -75,19 +81,23 @@ void GLFWModule::onProcessorNetworkEvaluationEnd() {
     // its work before we continue. This is needed to make sure that we have textures that are upto
     // data when we render the canvases.
     auto syncObj = glFenceSync(GL_SYNC_GPU_COMMANDS_COMPLETE, 0);
-    auto res = glClientWaitSync(syncObj, GL_SYNC_FLUSH_COMMANDS_BIT, 100000000);
+
+    const GLuint64 timeoutInNanoSec = 50'000'000;  // 50ms
+
+    auto res = glClientWaitSync(syncObj, GL_SYNC_FLUSH_COMMANDS_BIT, timeoutInNanoSec);
+    while (res == GL_TIMEOUT_EXPIRED) {
+        res = glClientWaitSync(syncObj, GL_SYNC_FLUSH_COMMANDS_BIT, timeoutInNanoSec);
```

Comment:
Should there be a max timeout here?

Reply:
maybe, this is more or less what we have in qt, seems to work fine there... usually windows will timeout after 30 sec anyway...